src/coreclr/src/jit/importer.cpp

```diff
@@ -106,7 +106,7 @@ void Compiler::impPushOnStack(GenTree* tree, typeInfo ti)
         // attempts to do that have proved too difficult.  Instead, we'll assume that in checks like this,
         // when there's a mismatch, it's because of this reason -- the typeInfo::AreEquivalentModuloNativeInt
         // method used in the last disjunct allows exactly this mismatch.
-        assert(ti.IsDead() || (ti.IsByRef() && (tree->TypeGet() == TYP_I_IMPL) || tree->TypeGet() == TYP_BYREF) ||
+        assert(ti.IsDead() || ((ti.IsByRef() && (tree->TypeGet()) == TYP_I_IMPL) || tree->TypeGet() == TYP_BYREF) ||
```

Comment:
It looks to be like this assert was incorrect before, but the new form makes it worse. I think this line should be:

`assert(ti.IsDead() || (ti.IsByRef() && ((tree->TypeGet() == TYP_I_IMPL) || (tree->TypeGet() == TYP_BYREF))) ||`

Reply:
Thank you, fixed.